### PR TITLE
Layers, Overlays, UI Changes, GeoJson, Snapping

### PR DESCRIPTION
**Pull request by webbertakken**
Added:
- ES5 compatibility in ESLint
- Configuring of TileLayers as BaseLayer or Overlay
- Configuring of WMS Layers as Overlay
- Configuring of GeoJson Layers as Overlay
- User Interface for switching between baseLayers and turning overlays on or off
- Marking GeoJson coordinates as circles on the map
- Updating GeoJson Layers when map position changes
- Configuring layers to snap the marker to its features on dragging

### CODE REVIEW

File: Form/Type/MapType.php [hunk 1]
```diff
@@ -101,8 +104,8 @@ public function buildView(FormView $view, FormInterface $form, array $options)
     {
         $view->vars['defaults']       = $options['defaults'];
         $view->vars['fields']         = $options['fields'];
-        $view->vars['map_dimensions'] = $options['map_dimensions'];
-        $view->vars['base_layer']     = $options['base_layer'];
+        $view->vars['map_dimensions'] = $options['mapDimensions'];
+        $view->vars['base_layers']     = $options['baseLayers'];
```
Comment (J063) [inline Form/Type/MapType.php]: surplus space before equals sign

**Comment by J063** [comment]
|`Resources/public/js/leaflet-embed.js:44`|
|---|
@{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}|@{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}

This variable name is not clear. (is it a button, field or something else?)

**Comment by J063** [comment]
|`Resources/public/js/leaflet-embed.js:274`|
|---|
@{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}|@{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}

Perhaps you could return 'layer' here as being undefined and remove the 'if' later on.

**Comment by J063** [comment]
|`Resources/public/js/leaflet-embed.js:415`|
|---|
@{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}|@{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}

Do we need a modal warning when one of the subsequent layers cannot be loaded? (as is done in initialiseBaseLayers using the modal structure)

**Comment by J063** [comment]
|`Resources/public/js/leaflet-embed.js:484`|
|---|
This is a possessive 'its' and not a contracted 'it has' or 'it is'

**Comment by webbertakken** [comment]
that would increase cyclomatic complexity, here i'm simply trying to be complete about what happens. 

(layer is undefined since it is defined at the top of the method)

**Comment by webbertakken** [comment]
I figured we wouldn't want a warning for every failing layer, since this is feedback t hat goes directly to the end-user as well.

File: Resources/public/js/leaflet-embed.js [hunk 2]
```diff
@@ -10,45 +10,131 @@ var CuriousMap = function (options) {
...
 
-  // Initialise Map
-  this.map = new L.Map(this.mapId);
-  this.initialiseBaseLayer(options.baseLayer);
-  this.initialiseOverlays(options.overlays);
+  // Initialise
+  this.initialise(options);
+};
+
+/*
```
Comment (J063) [inline Resources/public/js/leaflet-embed.js]: @{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}|@{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}|@{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}|@{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}|@{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}|@{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}|@{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}|@{c34f1ffd-de5f-44c1-b8f6-9ea92d34251a,Dick Webbink | Dation}

There seems to be an unused function on line 423 related to the commands performed on this line (74) and on line 13.

Either remove the individual commands or the unused function please.

**Comment by J063** [comment]
yes

**Comment by J063** [comment]
It indeed appears there is no added value for the customer in being annoyed with warnings of failing layers. Good point, leave it as it is.

**Pull request merged**